tests/FunctionAnyTest.php

```diff
@@ -2,18 +2,38 @@
 
 namespace React\Promise;
 
+use React\Promise\Exception\RangeException;
+
 class FunctionAnyTest extends TestCase
 {
     /** @test */
-    public function shouldResolveToNullWithEmptyInputArray()
+    public function shouldRejectWithRangeExceptionWithEmptyInputArray()
     {
         $mock = $this->createCallableMock();
         $mock
             ->expects($this->once())
             ->method('__invoke')
-            ->with($this->identicalTo(null));
+            ->with(
+                $this->callback(function($exception){
+                    return $exception instanceof RangeException &&
+                           'Input array must contain at least 1 item but contains only 0 items.' === $exception->getMessage();
```

Comment:
The documentation seems to suggest otherwise. IMO it makes sense to reject this with a `RangeException`, but we should probably also add this special case to the README.